tools/crm_mon.h

```diff
@@ -58,27 +58,26 @@ typedef enum mon_output_format_e {
     mon_output_cgi
 } mon_output_format_t;
 
-#define mon_show_times         (0x0001U)
-#define mon_show_stack         (0x0002U)
-#define mon_show_dc            (0x0004U)
-#define mon_show_count         (0x0008U)
-#define mon_show_nodes         (0x0010U)
-#define mon_show_resources     (0x0020U)
-#define mon_show_attributes    (0x0040U)
-#define mon_show_failcounts    (0x0080U)
-#define mon_show_operations    (0x0100U)
-#define mon_show_tickets       (0x0200U)
-#define mon_show_bans          (0x0400U)
-#define mon_show_fence_history (0x0800U)
-
-#define mon_show_headers       (mon_show_times | mon_show_stack | mon_show_dc \
-                               | mon_show_count)
-#define mon_show_default       (mon_show_headers | mon_show_nodes \
-                               | mon_show_resources)
-#define mon_show_all           (mon_show_default | mon_show_attributes \
-                               | mon_show_failcounts | mon_show_operations \
-                               | mon_show_tickets | mon_show_bans \
-                               | mon_show_fence_history)
+#define mon_show_stack          (1 << 0)
+#define mon_show_dc             (1 << 1)
+#define mon_show_times          (1 << 2)
+#define mon_show_counts         (1 << 3)
+#define mon_show_options        (1 << 4)
+#define mon_show_nodes          (1 << 5)
+#define mon_show_resources      (1 << 6)
+#define mon_show_attributes     (1 << 7)
+#define mon_show_failcounts     (1 << 8)
+#define mon_show_operations     (1 << 9)
+#define mon_show_fencing        (1 << 10)
```

Comment:
mon\_show\_fencing is an interesting one:

--fence-history can be 0 (show nothing), 1 (show the most recent failed fencing attempt for each node and all pending fencing operations, which is the default if --fence-history is not specified), 2 (as 1, and also show the most recent successful fencing for each node, which is the default if --fence-history is specified without a number), or 3 (show all fencing operations and not just the most recent).

Currently mon_show_fencing is turned on for everything but --fence-history 0, and the difference between 1/2/3 is determined by mon\_ops\_\* values rather than mon\_show\_\*. The 3 behavior makes sense as mon\_ops\_\*, but I wonder with this new option if failed/pending/successful history should be seen as separate sections, with --fence-history becoming a convenience alias (0 = --exclude fencing sections, 1 = --include pending and failed, 2 = --include all).

Reply:
Related to this - I can't tell you how many times I've pressed 0/1/2/3 in crm_mon interactive expecting it to do something.

Reply:
I think I might be handling fence history incorrectly at the moment.  I haven't found a file in cts/scheduler that results in any fencing being printed out.  I did originally add a mon_show_failures so I was thinking this could be more fine grained.  I'm just not sure how much we can get away with doing here without annoying users with changes too much.

Reply:
Fence history isn't stored in the cib. The only way to test it is with a live cluster.